equinox/nn/pool.py

```diff
@@ -256,3 +256,35 @@ def __init__(
             padding=padding,
             **kwargs,
         )
+
+
+class AdaptiveAvgPool1d(Module):
+    """Adaptive 1D downsampling for a target shape."""
+
+    target_size: int = static_field()
+
+    def __init__(self, target_size: int):
+        """**Arguments:**
+        - `target_size`: The target output size.
+        """
+
+        self.target_size = target_size
+
+    def __call__(self, x: Array):
+        assert (
+            x.ndim == 1
+        ), f"Only supports 1D input, received input with {x.ndim} dimensions."
```

Comment:
The convention in Equinox (and other Python projects) is for `assert` statements to be used for cases which should "never be hit". More precisely to indicate that the developer of the library has made a mistake somewhere, and that an assumed invariant isn't true.

In the case of user error, as here, then a `ValueError` is appropriate.